lib/internal/modules/cjs/loader.js

```diff
@@ -780,19 +780,19 @@ Module._load = function(request, parent, isMain) {
     }
   }
 
-  const filename = Module._resolveFilename(request, parent, isMain);
-  if (StringPrototypeStartsWith(filename, 'node:')) {
```

Comment:
Loading a builtin module prefixed with "node:" expects "ERR_UNKNOWN_BUILTIN_MODULE" to be thrown.
If the "resolve" function is called here, "MODULE_NOT_FOUND" will be thrown.